portal/src/main/webapp/js/src/study-view/component/PieChart.js

```diff
@@ -960,21 +967,22 @@ var PieChart = function(){
 
     //Called when the number of label biggen than 6, used by addPieLabels()
     function labelFunction() {
-        var _tableDiv = '';
+        var _tableDiv = [];
+        var _tableDivStr = '';
         var _labelLT = labelWLT; //label length threshold
         var _labelHeaderLT = labelHeaderWLT; //label header length threshold
 
-        _tableDiv += '<table';
+        _tableDiv.push('<table');
```

Comment:
instead of building up a string, I would suggest using _.template functionality together with the possibility to add new table rows via .append or appendTo functions in jquery. 

e.g. using jquery, something like this should work I think: 
var tableDiv = d3.select('#' + DIV.mainDiv).selectAll('.study-view-pie-label-copy');
$('<table />', {class: (category[0] === 'w1'? "smallTable" : "regularTable"}).appendTo(tableDiv);

etc